learn/references/java-interoperability/the-bindgen-tool.md

```diff
@@ -0,0 +1,223 @@
+Set the visibility modifier of the generated binding classes to public. This flag will be applicable only if the bindings are generated inside a single directory.
+
+`<class-name>...`
+One or more space-separated fully-qualified Java class names for which the Ballerina bridge code is to be generated. Please note that these class names should be provided at the end of the command.
+
+## Generated Bridge Code
+
+When the tool is run, a `.bal` file will be created to represent each Java class. This would contain the respective Ballerina binding class along with the required Java interoperability mappings. These `.bal` files will be generated inside separate modules representing the Java package structure by default. If the `[(-o|--output) <output-path>]` option is used, they will be generated inside a single directory.
+
+Apart from creating bindings for the specified Java classes, the command will also generate empty Ballerina binding classes for the dependent Java classes. A Java class would be considered dependent if it is used inside one of the generated Ballerina binding classes.
+
+A set of additional `.bal` files will be generated to store the error types used within the Ballerina binding classes.
+
+The generated bindings will be inside the specified output directory as follows.
+
+	<module-dir>
+	    ├── <class-name>.bal // generated classes
+        ├── ...
+	    ├── <class-name>.bal // generated dependent classes
+	    ├── ...
+	    ├── <class-name>.bal // generated error types
+	    └── ...
+
+## Java to Ballerina Mapping
+
+### Java Classes
+A Java class will be mapped to a Ballerina class. This Ballerina class will have the same name as the Java class.
+
+E.g., The generated Ballerina class of the `java.util.ArrayDeque` class will be as follows.
+```ballerina
+@java:Binding {
+    'class: "java.util.ArrayDeque"
+}
+distinct class ArrayDeque {
+
+    *java:JObject;
+
+    public handle jObj;
+
+    function init(handle obj) {
+        self.jObj = obj;
+    }
+
+    ...
+};
+```
+If there are multiple classes with the same simple name, they need to be generated using a single execution. The tool will then apply a numerical identifier at the end of duplicated class names. This could be manually changed into something meaningful if required.
+
+The format for specifying inner classes using the command is `<package-name>.ClassName$InnerClassName`. The dollar sign might have to be escaped using the backslash key.
+
+E.g., The command to generate bindings for `java.lang.Character.Subset` class will be as follows.
+```sh
+> bal bindgen java.lang.Character\$Subset
+```
+
+When referring a Java code to figure out the imported classes, you should be cautious about the Java classes from the `java.lang` package since these will not be visible as imports in the Java code. However, you need not generate bindings for the `java.lang.String` class since it is mapped into the Ballerina `string` type from within the Ballerina bindings generated.
```

Comment:
```suggestion
When referring a Java code to figure out the imported classes, you should be cautious about the Java classes from the `java.lang` package since these will not be visible as imports in the Java code. However, you need not generate bindings for the `java.lang.String` class since it is mapped into the Ballerina `string` type from within the generated Ballerina bindings.
```